test/e2e/checknodehealth_test.go

```diff
@@ -0,0 +1,382 @@
+		Expect(podStartupResult.Status).To(Equal(chmv1alpha1.CheckStatusUnhealthy))
+		Expect(podStartupResult.Message).To(ContainSubstring("timeout"))
+
+		By("Verifying StartedAt timestamp is set")
+		Expect(cnh.Status.StartedAt).NotTo(BeNil())
+
+		By("Verifying the timed out pod is cleaned up")
+		Eventually(func() int {
+			podList, err := clientset.CoreV1().Pods(checkerNamespace).List(ctx, metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", checknodehealth.CheckNodeHealthLabel, cnhName),
+			})
+			if err != nil {
+				GinkgoWriter.Printf("Failed to list pods: %v\n", err)
+				return -1
+			}
+			return len(podList.Items)
+		}, "30s", "2s").Should(Equal(0), "Timed out pod was not cleaned up within timeout")
+	})
+
+	It("should cleanup pod when CR is deleted", func() {
+		By("Creating a CheckNodeHealth CR with non-existent node")
+		cnhName = fmt.Sprintf("test-cnh-deletion-%d", time.Now().Unix())
+		nonExistentNode := "fake-node-for-deletion-test"
+		err := createCheckNodeHealthCR(ctx, k8sClient, cnhName, nonExistentNode)
+		Expect(err).NotTo(HaveOccurred())
+		GinkgoWriter.Printf("Created CheckNodeHealth CR: %s with non-existent node\n", cnhName)
+
+		By("Waiting for health check pod to be created and stuck in Pending")
+		Eventually(func() bool {
+			podList, err := clientset.CoreV1().Pods(checkerNamespace).List(ctx, metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", checknodehealth.CheckNodeHealthLabel, cnhName),
+			})
+			if err != nil || len(podList.Items) == 0 {
+				return false
+			}
+			pod := &podList.Items[0]
+			return pod.Spec.NodeName == nonExistentNode && pod.Status.Phase == corev1.PodPending
+		}, "10s", "1s").Should(BeTrue(), "Health check pod was not created or not in Pending state")
+
+		By("Deleting the CheckNodeHealth CR before timeout (within 5 seconds of creation)")
+		err = deleteCheckNodeHealthCR(ctx, k8sClient, cnhName)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Verifying the health check pod is deleted or terminating due to CR deletion")
+		Eventually(func() bool {
+			podList, err := clientset.CoreV1().Pods(checkerNamespace).List(ctx, metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", checknodehealth.CheckNodeHealthLabel, cnhName),
+			})
+			if err != nil {
+				return false
+			}
+			// Pod is considered cleaned up if it's deleted or all pods are in Terminating state
+			if len(podList.Items) == 0 {
+				return true
+			}
+			// Check if all pods are in Terminating state (DeletionTimestamp is set)
+			for _, pod := range podList.Items {
+				if pod.DeletionTimestamp == nil {
+					return false
+				}
+			}
+			return true
+		}, "30s", "2s").Should(BeTrue(), "Health check pod was not deleted or terminating within timeout")
+
+		By("Verifying the CheckNodeHealth CR is deleted")
+		Eventually(func() bool {
+			return !checkNodeHealthCRExists(ctx, k8sClient, cnhName)
+		}, "30s", "1s").Should(BeTrue(), "CheckNodeHealth CR was not deleted within timeout")
+
```

Comment:
The CR has already been confirmed deleted? The CR is deleted earlier in a call to `deleteCheckNodeHealthCR` and there is an assertion there was no error? I don't think this is necessary.

Reply:
you are right, fixed